assets/js/components/ExecutionResults/ExecutionResults.jsx

```diff
@@ -0,0 +1,154 @@
+  cloudProvider,
+  hostnames = [],
+  onExecutionFetch = getExecutionResult,
+  onCatalogFetch = getCatalog,
+  onCatalogRefresh = () => {},
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [executionData, setExecutionData] = useState(null);
+  const [catalog, setCatalog] = useState(null);
+  const [selectedCheck, setSelectedCheck] = useState(null);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    Promise.all([onExecutionFetch(executionID), onCatalogFetch()])
+      .then(
+        ([{ data: fetchedExecutionData }, { data: fetchedCatalogData }]) => {
+          setLoading(false);
+          setExecutionData(fetchedExecutionData);
+          setCatalog(fetchedCatalogData.items);
+        }
+      )
+      .catch((error) => {
+        setLoading(false);
+        logError(error);
+      });
+  }, [onExecutionFetch, onCatalogFetch, setExecutionData, setCatalog]);
+
+  if (loading) {
+    return <LoadingBox text="Loading checks execution..." />;
+  }
+
+  if (executionData?.status === 'running') {
+    return <LoadingBox text="Check execution currently running..." />;
+  }
+
+  const checkResults = getCheckResults(executionData);
+  const hosts = getHosts(checkResults);
+  const checks = getChecks(checkResults);
+
+  return (
+    <div>
+      <Modal
+        title={catalog?.find(({ id }) => id === selectedCheck)?.description}
+        open={modalOpen}
+        onClose={() => setModalOpen(false)}
+      >
+        <ReactMarkdown className="markdown" remarkPlugins={[remarkGfm]}>
+          {catalog?.find(({ id }) => id === selectedCheck)?.description}
+        </ReactMarkdown>
+      </Modal>
+      <BackButton url={`/clusters/${clusterID}`}>
+        Back to Cluster Details
+      </BackButton>
+      <div className="flex mb-4 justify-between">
+        <h1 className="text-3xl w-3/5">
+          <span className="font-medium">Checks Results for cluster</span>{' '}
+          <span
+            className={classNames('font-bold', truncatedClusterNameClasses)}
+          >
+            {clusterName}
+          </span>
+        </h1>
+      </div>
+      {cloudProvider == UNKNOWN_PROVIDER && (
+        <WarningBanner>
+          The following results are valid for on-premise bare metal platforms.
+          <br />
+          If you are running your HANA cluster on a different platform, please
+          use results with caution
+        </WarningBanner>
+      )}
+      <ResultsContainer
+        catalogError={false}
+        clusterID={clusterID}
+        hasAlreadyChecksResults
+        selectedChecks={checks}
+        onCatalogRefresh={onCatalogRefresh}
+      >
+        {hosts &&
+          hosts.map((agentID, idx) => (
+            <HostResultsWrapper
+              key={idx}
+              hostname={hostnames.find(({ id }) => agentID === id)?.hostname}
+              reachable
+              unreachableMessage=""
+            >
+              {checks.map((checkID) => {
```

Comment:
Maybe we can extract that logic, inside the arrow function, inside another component? Seems a little bit convoluted to follow